webrender/src/frame.rs

```diff
@@ -442,13 +443,27 @@ impl<'a> FlattenContext<'a> {
             SpecificDisplayItem::Rectangle(ref info) => {
                 if !self.try_to_add_rectangle_splitting_on_clip(
                     &prim_info,
-                    &info.color,
+                    &RectangleContent::Fill(info.color),
                     &clip_and_scroll,
                 ) {
                     self.builder.add_solid_rectangle(
                         clip_and_scroll,
                         &prim_info,
-                        &info.color,
+                        &RectangleContent::Fill(info.color),
+                        PrimitiveFlags::None,
+                    );
+                }
+            }
+            SpecificDisplayItem::ClearRectangle => {
+                if !self.try_to_add_rectangle_splitting_on_clip(
```

Comment:
Let's not bother trying to split clear rectangles - we can just directly call ```add_solid_rectangle``` here.